php/EE/Migration/Containers.php

```diff
@@ -0,0 +1,183 @@
+				'EE\Migration\Containers::nginxproxy_container_down',
+				null,
+				[ $existing_nginx_proxy_image ]
+			);
+		}
+
+		// Upgrade cron container
+		$existing_cron_image = EE::launch( 'docker inspect --format=\'{{.Config.Image}}\' ee-cron-scheduler', false, true );
+		if ( 0 === $existing_cron_image->return_code ) {
+			self::$rsp->add_step(
+				'upgrade-cron-container',
+				'EE\Migration\Containers::cron_container_up',
+				'EE\Migration\Containers::cron_container_down',
+				null,
+				[ $existing_cron_image ]
+			);
+		}
+	}
+
+	public static function nginxproxy_container_up() {
+		$EE_CONF_ROOT      = EE_CONF_ROOT;
+		$nginx_proxy_image = 'easyengine/nginx-proxy:v' . EE_VERSION ;
+		$ee_proxy_command = "docker run --name ee-nginx-proxy -e LOCAL_USER_ID=`id -u` -e LOCAL_GROUP_ID=`id -g` --restart=always -d -p 80:80 -p 443:443 -v $EE_CONF_ROOT/nginx/certs:/etc/nginx/certs -v $EE_CONF_ROOT/nginx/dhparam:/etc/nginx/dhparam -v $EE_CONF_ROOT/nginx/conf.d:/etc/nginx/conf.d -v $EE_CONF_ROOT/nginx/htpasswd:/etc/nginx/htpasswd -v $EE_CONF_ROOT/nginx/vhost.d:/etc/nginx/vhost.d -v /var/run/docker.sock:/tmp/docker.sock:ro -v $EE_CONF_ROOT:/app/ee4 -v /usr/share/nginx/html $nginx_proxy_image" ;
+
+		default_launch( 'docker rm -f ee-nginx-proxy', false, true );
+
+		if ( ! default_launch( $ee_proxy_command, false, true ) ) {
+			throw new \Exception( ' Unable to upgrade ee-nginx-proxy container' );
+		}
+
+	}
+
+	public static function nginxproxy_container_down( $existing_nginx_proxy_image ) {
+		$EE_CONF_ROOT      = EE_CONF_ROOT;
+		$nginx_proxy_image = trim( $existing_nginx_proxy_image->stout );
+		$ee_proxy_command = "docker run --name ee-nginx-proxy -e LOCAL_USER_ID=`id -u` -e LOCAL_GROUP_ID=`id -g` --restart=always -d -p 80:80 -p 443:443 -v $EE_CONF_ROOT/nginx/certs:/etc/nginx/certs -v $EE_CONF_ROOT/nginx/dhparam:/etc/nginx/dhparam -v $EE_CONF_ROOT/nginx/conf.d:/etc/nginx/conf.d -v $EE_CONF_ROOT/nginx/htpasswd:/etc/nginx/htpasswd -v $EE_CONF_ROOT/nginx/vhost.d:/etc/nginx/vhost.d -v /var/run/docker.sock:/tmp/docker.sock:ro -v $EE_CONF_ROOT:/app/ee4 -v /usr/share/nginx/html $nginx_proxy_image" ;
+
+		default_launch( 'docker rm -f ee-nginx-proxy', false, true );
+
+		if ( ! default_launch( $ee_proxy_command, false, true ) ) {
+			throw new \Exception( ' Unable to restore ee-nginx-proxy container' );
+		}
+	}
+
+	public static function cron_container_up() {
+		$cron_image = 'easyengine/cron:v' . EE_VERSION;
+		$cron_scheduler_run_command = 'docker run --name ee-cron-scheduler --restart=always -d -v ' . EE_CONF_ROOT . '/cron:/etc/ofelia:ro -v /var/run/docker.sock:/var/run/docker.sock:ro ' . $cron_image ;
+
+		default_launch( 'docker rm -f ee-cron-scheduler', false, true );
+
+		if ( ! default_launch( $cron_scheduler_run_command, false, true ) ) {
+			throw new \Exception( ' Unable to upgrade ee-cron-scheduler container' );
+		}
+	}
+	public static function cron_container_down( $existing_cron_image ) {
```

Comment:
Add a line before creating a new function as you've added for other functions. Please maintain code consistency.